itou/users/management/commands/certify_selected_administrative_criteria.py

```diff
@@ -0,0 +1,152 @@
+import concurrent
+import logging
+from math import ceil
+
+from dateutil.relativedelta import relativedelta
+from django.db.models import Q
+from django.utils import timezone
+
+from itou.eligibility.models.geiq import GEIQAdministrativeCriteria, GEIQSelectedAdministrativeCriteria
+from itou.eligibility.models.iae import AdministrativeCriteria, SelectedAdministrativeCriteria
+from itou.users.models import User
+from itou.utils.command import BaseCommand
+from itou.utils.iterators import chunks
+
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Certify selected administrative criteria from eligiility diagnosis made during the last 6 months
+    by calling the Particulier API."""
+
+    def add_arguments(self, parser):
+        parser.add_argument("--limit", dest="limit", action="store", type=int)
+        parser.add_argument("--synchronously", dest="synchronously", action="store_true")
+        parser.add_argument("--verbose", dest="verbose", action="store_true")
+        parser.add_argument("--wet-run", dest="wet_run", action="store_true")
+
+    def call_api_and_store_result(
+        self,
+        SelectedAdministrativeCriteriaModel,
+        AdministrativeCriteriaModel,
+        limit=None,
+        synchronously=False,
+        verbose=False,
+        wet_run=False,
+    ):
+        if not verbose:
+            # Don't log HTTP requests detail.
+            logging.getLogger("httpx").setLevel(logging.WARNING)
+            logging.getLogger("httpcore").setLevel(logging.WARNING)
+
+        total_criteria = 0
+        total_criteria_with_certification = 0
+        found_beneficiaries = set()
+        found_not_beneficiaries = set()
+        not_found_users = set()  # 404
+        server_errors = 0  # 429, 503, 504
+
+        criteria = AdministrativeCriteriaModel.objects.certifiable()
+        six_months = timezone.now() - relativedelta(months=6)
+        criteria_pks_qs = (
+            SelectedAdministrativeCriteriaModel.objects.filter(
+                administrative_criteria__in=criteria,
+                eligibility_diagnosis__created_at__gte=six_months,
+                eligibility_diagnosis__job_seeker__jobseeker_profile__birth_country__isnull=False,
+                eligibility_diagnosis__job_seeker__jobseeker_profile__birthdate__isnull=False,
+                eligibility_diagnosis__job_seeker__first_name__isnull=False,
+                eligibility_diagnosis__job_seeker__last_name__isnull=False,
+                eligibility_diagnosis__job_seeker__title__isnull=False,
+            )
+            .exclude(Q(certified__isnull=False) | Q(data_returned_by_api__error__contains="not_found"))  # exclude 404
+            .values("pk", "eligibility_diagnosis__job_seeker__pk")
+        )
+        if limit:
+            criteria_pks_qs = criteria_pks_qs[:limit]
+        criteria_pks = list([val["pk"] for val in criteria_pks_qs])
```

Comment:
Tu peux récupérer les pks avec `zip()`

```suggestion
            .values_list("pk", "eligibility_diagnosis__job_seeker__pk")
        )
        if limit:
            criteria_pks_qs = criteria_pks_qs[:limit]
        criteria_pks, users_pks = zip(*criteria_pks_qs)
```
Par contre il faut mettre le early return avant, car ça plante aussi quand le queryset est vide